client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java

```diff
@@ -507,7 +507,17 @@ public boolean isValid(int timeout)
         if (timeout < 0) {
             throw new SQLException("Timeout is negative");
         }
-        return !isClosed();
+
+        if (isClosed()) {
+            return false;
+        }
+
+        try (TrinoStatement statement = doCreateStatement()) {
+            return statement.internalExecute("SELECT 1");
```

Comment:
`internalExecute` returns boolean saying whether there is a result set
so it's not valid response for `isValid` method.
also, the result set initiated here isn't getting closed.

Must do changes:

- close the result set
- return true unless there was a failure

IMO good to do changes:

- use a dedicated endpoint, avoiding polluting the query history; this can be done as a follow up, when we fix getConnection to make it validate stuff

Reply:
Can we get this PR back and merge it? It's worth fixing JDBC compliance issues before getting a dedicated endpoint.

Require changes:

- Call`TrinoConnection#isValid(int timeout)` in `NonRegisteringTrinoDriver#connect(String url, Properties info)` to fix #13341 and #8542